src/qir2qasm_trans/qir_trans/builder.py

```diff
@@ -0,0 +1,611 @@
+import re
+from collections import OrderedDict
+from dataclasses import dataclass
+from typing import List, Optional, Tuple, Union
+
+from llvmlite.binding.module import ValueRef
+from llvmlite.binding.typeref import TypeRef
+from llvmlite.ir import FunctionType, IdentifiedStructType
+from openqasm3 import ast
+
+
+class FunctionBuilder:
+    def building(
+        self, symbols, ret_type: TypeRef, operands: List[ValueRef]
+    ) -> Tuple[Optional[Union[ast.IndexedIdentifier, ast.Identifier]], List[ast.Statement]]:
+        return None, []
+
+
+@dataclass
+class FunctionInfo:
+    type: FunctionType
+    def_statement: Optional[ast.QuantumGateDefinition]
+    builder: FunctionBuilder
+
+
+class InstructionBuilder:
+    def building(
+        self,
+        **kwargs,
+    ) -> Optional[ast.IndexedIdentifier]:
+        return None
+
+
+@dataclass
+class InstructionInfo:
+    builder: InstructionBuilder
+
+
+class DeclBuilder:
+    def building(self, name: str, size: int) -> ast.Statement:
+        return None
+
+
+@dataclass
+class StructureInfo:
+    type_qir: IdentifiedStructType
+    type_ast: Union[ast.ClassicalType, str]
+    decl_builder: DeclBuilder
+    # call_builder
+
+
+@dataclass
+class BranchInfo:
+    br_condition: Optional[ast.Expression]
```

Comment:
Can we use a more explicit naming convention? For example, use "branch_condition" instead of "br_condition"